gems/aws-sdk-core/lib/aws-sdk-core/plugins/regional_endpoint.rb

```diff
@@ -47,21 +43,27 @@ class RegionalEndpoint < Seahorse::Client::Plugin
       end
 
       def after_initialize(client)
-        if client.config.region.nil? or client.config.region == ''
+        if client.config.region.nil? || client.config.region == ''
           raise Errors::MissingRegionError
         end
+
+        # check region format follows pattern <continent>-<direction>-<number>
+        unless client.config.region =~ /^[a-zA-Z]([a-zA-Z0-9\-]*[a-zA-Z0-9])?$/
```

Comment:
Could shorten this by either: making it case insensitive or using bracketed character classes.  eg: [:alpha:] and [:alnum:]

Reply:
Pending whatever the teams decide...